pkg/lib/indexer/indexer.go

```diff
@@ -513,38 +523,72 @@ func (i ImageIndexer) ExportFromIndex(request ExportFromIndexRequest) error {
 	}
 
 	var errs []error
+	var g errgroup.Group
+	var mu = &sync.Mutex{}
+
+	concurrentGoroutines := make(chan struct{}, concurrencyLimitForExport)
+
 	for _, bundleImage := range bundles {
-		// try to name the folder
-		folderName, err := dbQuerier.GetBundleVersion(context.TODO(), bundleImage)
-		if err != nil {
+		bundleImage := bundleImage // to ensure every go routine gets a it's own copy of closure variable
+
+		g.Go(func() error {
+			concurrentGoroutines <- struct{}{}
+			defer func() { <-concurrentGoroutines }()
+
+			// get the bundle version to be used in naming the folder
+			bundleVersion, err := dbQuerier.GetBundleVersion(context.TODO(), bundleImage)
+			if err != nil {
+				i.Logger.Errorf("error getting bundle version from image: %s", err)
+				return err
+			}
+			if bundleVersion == "" {
+				// operator-registry does not care about the folder name
+				bundleVersion = bundleImage
+			}
+			// fetch bundle name from bundle image
+			pkgName := strings.FieldsFunc(bundleImage, splitBundlePath)[2]
+			exporter := bundle.NewExporterForBundle(bundleImage, filepath.Join(request.DownloadPath, pkgName, bundleVersion), request.ContainerTool)
+			if err := exporter.Export(); err != nil {
```

Comment:
I noticed that you removed the cancellation behavior from an earlier revision. You might consider using https://pkg.go.dev/golang.org/x/sync/errgroup?tab=doc#WithContext, and propagating it to `Export()` (and further to https://github.com/operator-framework/operator-registry/blob/master/pkg/lib/bundle/exporter.go#L62 and friends) so that things like image pulls can be interrupted.

Reply:
Not using `errgroup` since the team's input was to collect errors instead of failing it on the first `err`. Switched back to `sync.WaitGroup`

Reply:
Ok. That makes sense if the idea is to allow everything to complete despite any errors.